cmake/SetupHdf5.cmake

```diff
@@ -52,6 +52,34 @@ if(HDF5_IS_PARALLEL)
   endif()
 endif()
 
+if(HDF5_USE_STATIC_LIBRARIES)
+  # If we are using static libraries for HDF5, try to use static libraries
+  # for the libsz and transitive libaec dependency. These aren't very
+  # common, while other dependencies like libz are, so we leave those
+  # as shared library dependencies.
+  get_target_property(
+    _HDF5_INTERFACE_LINK_LIBS HDF5::HDF5 INTERFACE_LINK_LIBRARIES)
+  string(FIND
+    "${_HDF5_INTERFACE_LINK_LIBS}" "libsz.so" _LOCATION_OF_LIBSZ)
+  if (NOT ${_LOCATION_OF_LIBSZ} EQUAL -1)
+    find_library(_libsz NAMES libsz.a)
```

Comment:
libsz is an optional dependency of hdf5 that you specify when building hdf5. Also when building hdf5 you can specify if you want to build static libs, shared libs, or both. So if a static build of everything is important, can we just make sure to build hdf5 in an all-static way instead of trying to patch transitive dependencies in the spectre build system? For an all-static build we have to control the environment very carefully anyway, probably in a container, so we can make sure to build hdf5 correctly as well.

Reply:
I'm not sure what you're looking for in terms of changes or are suggesting. This works in the container for producing a portable statically linked executable. I linked against static HDF5 and then found out that `libsz` was linked dynamically for that. Are you suggesting we compile HDF5 ourselves statically in the container?

Reply:
Yes that's what I thought, in fact I thought we had to build a modified container where everything is statically linked anyway?